sympy/series/formal.py

```diff
@@ -1143,6 +1144,87 @@ def integrate(self, x=None, **kwargs):
 
         return self.func(f, self.x, self.x0, self.dir, (ak, self.xk, ind))
 
+    def convolve(self, other, x=None, n=6, cycle=0, dyadic=False, subset=False):
```

Comment:
I think this api might not be the best for the task. If I understand correctly only one of ``dyadic`` or ``subset`` can be set at  a time or both can be kept ``False``. Does it make sense to change it something like ``method='dyadic'`` or ``method=None`` instead of having separate kwarg for each of the methods. Also how about supporting other convolution methods allowed by sympy's ``convolution``?

Reply:
That can be done. So the different methods would be `linear`, `dyadic` and `subset`. The reason I removed `dps` and `prime` is because I felt that they are not of much relevance in formal power series. Also they aren't methods. We can decide on that and keep it as per discussion.

Reply:
> That can be done. So the different methods would be `linear`, `dyadic` and `subset`. The reason I removed `dps` and `prime` is because I felt that they are not of much relevance in formal power series. Also they aren't methods. We can decide on that and keep it as per discussion.

My concern is currently a user can do ``f1.convolve(f2, dyadic=True, subset=True)``, which shouldn't be possible right? This can be solved if we just have one ``kwarg`` instead.

Reply:
The `method` has been implemented. ping @leosartaj